lib/Doctrine/DBAL/Tools/Console/ConsoleRunner.php

```diff
@@ -30,20 +36,30 @@ public static function createHelperSet(Connection $connection)
     }
 
     /**
-     * Runs console with the given helperset.
+     * Runs console with the given connection provider or helperset (deprecated).
      *
-     * @param Command[] $commands
+     * @param ConnectionProvider|HelperSet $helperSetOrConnectionProvider
+     * @param Command[]                    $commands
      *
      * @return void
      */
-    public static function run(HelperSet $helperSet, $commands = [])
+    public static function run($helperSetOrConnectionProvider, $commands = [])
     {
         $cli = new Application('Doctrine Command Line Interface', Version::VERSION);
 
         $cli->setCatchExceptions(true);
-        $cli->setHelperSet($helperSet);
 
-        self::addCommands($cli);
+        /** @var ConnectionProvider|null $connectionProvider */
+        $connectionProvider = $helperSetOrConnectionProvider;
+        if ($helperSetOrConnectionProvider instanceof HelperSet) {
+            $connectionProvider = null;
+            $cli->setHelperSet($helperSetOrConnectionProvider);
+            @trigger_error(sprintf('Passing an instance of "%s" as the first argument is deprecated. Pass an instance of "%s" instead.', HelperSet::class, ConnectionProvider::class), E_USER_DEPRECATED);
```

Comment:
Should `trigger_error()` happen in the very beginning of the `if`-branch? Even if `$cli->setHelperSet()` throws an exception (regardless of whether it's possible), we want the deprecation to be triggered.